From c50dab84e0c911f7b89532700fca27d42a77b3c1 Mon Sep 17 00:00:00 2001
From: kxxt <rsworktech@outlook.com>
Date: Fri, 13 Sep 2024 12:48:44 +0000
Subject: [PATCH] cargo_crate.gni: add riscv64 and loong64 cases

This CL syncs the condition of 64 bit pointer with
https://github.com/chromium/chromium/blob/31f058a669fd063cab3439df5c877ed3005d9832/base/allocator/partition_allocator/partition_alloc.gni#L21-L24

Bug: none
Change-Id: Id2f1cfaa0c5167fb61c74505aba03716337b071f
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5850670
Commit-Queue: Nico Weber <thakis@chromium.org>
Reviewed-by: Nico Weber <thakis@chromium.org>
Reviewed-by: Arthur Eubanks <aeubanks@google.com>
Cr-Commit-Position: refs/heads/main@{#1355138}
---

diff --git a/build/rust/cargo_crate.gni b/build/rust/cargo_crate.gni
index e5d1b2f8..7290451 100644
--- a/build/rust/cargo_crate.gni
+++ b/build/rust/cargo_crate.gni
@@ -429,7 +429,8 @@
           rust_abi_target,
         ]
       }
-      if (current_cpu == "arm64" || current_cpu == "x64") {
+      if (current_cpu == "arm64" || current_cpu == "x64" ||
+          current_cpu == "loong64" || current_cpu == "riscv64") {
         args += [
           "--pointer-width",
           "64",
